packages/console/src/pages/EnterpriseSsoDetails/Settings/index.tsx

```diff
@@ -0,0 +1,266 @@
+import {
+  type SsoConnector,
+  type SsoConnectorWithProviderConfig,
+  type RequestErrorBody,
+  findDuplicatedOrBlockedEmailDomains,
+} from '@logto/schemas';
+import { generateStandardShortId } from '@logto/shared/universal';
+import { conditional, conditionalArray, conditionalString } from '@silverhand/essentials';
+import { t as globalTranslate } from 'i18next';
+import { HTTPError } from 'ky';
+import { useForm, Controller, FormProvider } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+
+import DetailsForm from '@/components/DetailsForm';
+import FormCard from '@/components/FormCard';
+import UnsavedChangesAlertModal from '@/components/UnsavedChangesAlertModal';
+import FormField from '@/ds-components/FormField';
+import Select from '@/ds-components/Select';
+import TextInput from '@/ds-components/TextInput';
+import MultiInput, { type Option as MultiInputOption } from '@/ds-components/TextInput/MultiInput';
+import useApi from '@/hooks/use-api';
+import useUserAssetsService from '@/hooks/use-user-assets-service';
+import { SyncProfileMode } from '@/types/connector';
+import { trySubmitSafe } from '@/utils/form';
+import { uriValidator } from '@/utils/validator';
+
+import LogosUploader from './LogosUploader';
+import * as styles from './index.module.scss';
+
+type Props = {
+  isDeleted: boolean;
+  data: SsoConnectorWithProviderConfig;
+  onUpdated: (data: SsoConnectorWithProviderConfig) => void;
+};
+
+export type FormType = Pick<SsoConnector, 'branding' | 'connectorName'> & {
+  syncProfile: SyncProfileMode;
+  domains: MultiInputOption[];
+};
+
+const duplicatedDomainsErrorCode = 'single_sign_on.duplicated_domains';
+const forbiddenDomainsErrorCode = 'single_sign_on.forbidden_domains';
+
+const dataToFormParser = (data: SsoConnectorWithProviderConfig) => {
+  const { branding, connectorName, domains, syncProfile } = data;
+  return {
+    branding,
+    connectorName,
```

Comment:
These two don't have to be deconstructed from `data`, can use `...rest` instead

Reply:
We only show picked fields of data not all data.